### ISSUE TITLE
Possible bug ? desktop.rdp  vs  application.rdp

### ISSUE THREAD

**Issue by Skorium** [microsoft/RdpGamepad#21]
Hi,

I'm not sure if this is a bug or if it works as intended.
I have 2 rdp files:

- The 1st, let's call it "desktop.rdp", is starting the remote connection to full desktop access and here the controller Xbox 360 is recognised without problem. I can start Dolphin/RPCS3 and use the control via the remote desktop session without problems.

- The 2nd, let's call it "application.rdp", connects to the same remote desktop but the rdp file was edited to start only one app remotely (so it starts RDP and only display the window of that specific app). It is set to start the application RPCS3 but when I use this "application.rdp" the Xbox 360 controller aren't detected in RPCS3 (remember they are working when using "desktop.rdp" on the same remote computer and client). I haven't tried with Dolphin setup the same way as it didn't work with RPCS3 but that can be done easily :)

I can share the .rdp files if required. They are identical except one start a remote application via configuration "remoteapplicationprogram:s:C:\Path\To\RPCS3\rpcs3.exe"

Any clues at why 2 different outcome here, is it possible to get that to work ? More tests to do on my end ? Or do you need more details, if so what do you need ?

Thanks in advance for reading me :)

Skorium

**Comment by jpflouret** [comment]
This is probably a timing issue. When you first connect to the remote computer, it takes a second or two for the gamepad to be available. If you launch a program before that, it will likley not find any available gamepads yet.

You can verify that the gamepad connection is established using the [latest release](https://github.com/microsoft/RdpGamepad/releases/tag/v1.0.2) of the reciever app (v1.0.2). If you hover over the white Xbox gamepad icon on the system notification area and read the tool tip from the receiver program it should say "Microsoft Remote Gamepad Receiver - Connected" if the gamepad is available. If it says "... - Disconnected" then something is interfering with the gamepad connection.

If it is a timing issue, you can work around it by writing a batch file that uses the timeout command to introduce a delay before launching any program that needs the gamepads available. Something maybe like this:

    @echo off
    timeout /t 5
    C:\Path\To\RPCS3\rpcs3.exe

Let me know if that fixes your issue.

**Comment by Skorium** [comment]
So tried this batch file, up to 15s. The gamepad is still not connecting using the "application.rdp" method above. Via full remote desktop access no issues at all, gamepad works

I found some sort of workaround for now with the batch file. I disabled the automatic starts of the receiver and start it with the batch file then it pickup the gamepad.

```
@echo off
"C:\Program Files\Microsoft Remote Desktop Gamepad Receiver\RdpGamepadViGEm64.exe"
timeout 5 > nul
START rpcs3.exe
```

Could it be related to some memory/process protection of some sorts ?

PS: Already on the latest version of every part of the setup but thanks for the suggestion 👍
